content/push/publish.textile

```diff
@@ -0,0 +1,768 @@
+    @"notification": @{
+        @"title": @"Hello from Ably!",
+        @"body": @"Example push notification from Ably."
+    },
+    @"data": @{
+        @"foo": @"bar",
+        @"baz": @"qux"
+    }
+};
+[rest.push.admin publish:recipient data:data callback:^(ARTErrorInfo *error)
+```
+
+```[swift]
+let recipient: [String: Any] = [
+    "transportType": "apns",
+    "deviceToken": "XXXXXXXX"
+]
+
+let data: [String: Any] = [
+    "notification": [
+        "title": "Hello from Ably!",
+        "body": "Example push notification from Ably."
+    ],
+    "data": [
+        "foo": "bar",
+        "baz": "qux"
+    ]
+]
+rest.push.admin.publish(recipient, data: data)
+```
+
+```[java]
+Message message = new Message("example", "rest data");
+message.extras = io.ably.lib.util.JsonUtils.object()
+    .add("push", io.ably.lib.util.JsonUtils.object()
+        .add("notification", io.ably.lib.util.JsonUtils.object()
+            .add("title", "Hello from Ably!")
+            .add("body", "Example push notification from Ably."))
+        .add("data", io.ably.lib.util.JsonUtils.object()
+            .add("foo", "bar")
+            .add("baz", "qux")));
+
+rest.push.admin.publish(arrayOf(Param("transportType", "apns"), Param("deviceToken", deviceToken)), message);
+```
+
+```[python]
+recipient = {'transportType': 'apns', 'deviceToken': 'XXXXXXX'}
+message = {
+  'push': {
+    'notification': {
+      'title': 'Hello from Ably!',
+      'body': 'Example push notification from Ably.'
+    }
+  }
+}
+
+rest.push.admin.publish(recipient, message)
+```
+
+```[php]
+$recipient = [  'transportType' => 'apns', 'deviceToken' => 'XXXXXXX' ];
+$data = [ 'push' =>
+          [ 'notification' =>
+            [ 'title' => 'Hello from Ably!',
+              'body' => 'Example push notification from Ably.'
+            ]
+          ]
+        ];
+$rest->push->admin->publish( $recipient, $data );
+```
+
+```[csharp]
+var recipient = new {
+    deviceId = "xxxxxxxxxxx"
+};
+var data = new {
+    notification = new {
+        title = "Hello from Ably!",
+        body = "Example push notification from Ably."
+    }
+};
+
+rest.push.admin.publish(recipient, data);
+```
+
+```[flutter]
+    'deviceId': 'xxxxxxxxxxx',
+};
+var data = {
+    'notification': {
+        'title': 'Hello from Ably!',
+        'body': 'Example push notification from Ably.',
+    },
+};
+
+rest.push.admin.publish(recipient, data);
+
+```
+
+
+h2(#via-channels). Publish via channels
+
+Publishing via channels is modeled on Ably’s "channel":#channels infrastructure, facilitating the delivery of push notifications across a network of subscribed devices. This method publishes messages through predefined channels, which devices must "subscribe":#sub-channels to to receive updates. This process ensures registered devices in the specified channels recieve the correct push notifications. Publishing via channels is particularly useful for publishing notifications to multiple groups with varying privileges.
```

Comment:
I think it may be worth highlighting that it's a different subscribe than just receiving ordinary messages. So you have to subscribe explicitly to receive push notifications on that channel.